packages/tpetra/core/src/Tpetra_Details_DefaultTypes.hpp

```diff
@@ -126,7 +126,7 @@ namespace DefaultTypes {
 
 #ifdef KOKKOS_ENABLE_HIP
   template<>
-  struct CommBufferMemorySpace<Kokkos::Experimental::HIPHostPinnedSpace>
+  struct CommBufferMemorySpace<Kokkos::Experimental::HIP>
   {
     using type = Kokkos::Experimental::HIPHostPinnedSpace;
```

Comment:
It should be HIPHostPinned for now, until we know that HIP-aware MPI support is good on the testbeds. In the long term, you're right, it will be HIPSpace (like CudaSpace is used for Cuda).

Reply:
OK. I thought GPU-aware MPI is handled somewhere else and by default host buffers are used. Am I wrong here? In that case I should also change to an host-accessible memory space in #9086.

Reply:
@masterleinad Yeah, SYCL should use a host-accessible space for now too. Right now there's only logic throughout the code for CUDA-aware MPI specifically. There will be more work ahead to make it general across HIP/SYCL/OpenMPTarget.

Reply:
OK. I switched to `SYCLSharedUSMSpace` in [`d69c5f4` (#9086)](https://github.com/trilinos/Trilinos/pull/9086/commits/d69c5f457896113d48a385b4b16b70be620c983f).

Reply:
I'm not sure Pinned memory (for comm) is the right for testing out HIP w/tpetra.  That will trigger non-standard code paths - we want to exercise HIPSpace primarily - and for  Comm, then set `TPETRA_ASSUME_CUDA_AWARE_MPI=0` (or however it's spelled) - that's a better path imo.  This would mimic how Cuda was R&D'd - Tpetra *never* used UVM w/MPI - Mark was adamant about that.

Furthermore, app teams started messing around with Pinned, because they thought it would improve performance (it generally didn't - and their motivation was based on a lack of understanding of the MPI runtime (i.e., we had cuda-aware MPI ON, which is inherently slower).

When we had the discussion about the FECrs error - I wasn't aware that the default comm buffer was being put as Pinned... We should determine if the error remains if Tpetra is not cuda-aware and the comm buffer is device memory.

Reply:
@jjellio OK, that's a good idea. I'll see how much is needed to put in HIPSpace (maybe not much, ``Behavior::assumeMpiIsCudaAware()`` is only called in a dozen places or so).

Reply:
@masterleinad Did you see James's comment here? It applies to the SYCL PR as well. I thought there would be issues with MPI CUDA-aware codepaths being taken, but actually Tpetra decides at configure time that since CUDA is not enabled, MPI is not CUDA-aware. It's kind of doing the right thing for the wrong reason in HIP. But it will work until "cuda-aware" options are replaced with portable  "device-aware MPI". I switched CommBufferMemorySpace to HIPSpace and all tests still passed with no other changes.

Reply:
OK. I wasn't quite sure if there is more needed to make it work in an expected way apart from switching the memory space again. Anyway, I reverted the corresponding commit, see [`8b59e2f` (#9086)](https://github.com/trilinos/Trilinos/pull/9086/commits/8b59e2f89e13d63f18dd278df5e53bfc72b11bc9).